dbms/src/Core/BlockUtils.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Core/BlockUtils.h>
+
+namespace DB
+{
+bool blockEqual(const Block & expected, const Block & actual, String & unequal_msg)
```

Comment:
It has a "gtest-specified" version in `FunctionTestUtils.cpp` 😂

Reply:
> It has a "gtest-specified" version in `FunctionTestUtils.cpp` 😂

I will not modify the code, since dbgNaturalDag that use the utils is not used for now.